shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/GenotypeGvcfs.scala

```diff
@@ -0,0 +1,78 @@
+package nl.lumc.sasc.biopet.pipelines.shiva
+
+import java.io.File
+
+import nl.lumc.sasc.biopet.core.BiopetQScript.InputFile
+import nl.lumc.sasc.biopet.core.{BiopetQScript, PipelineCommand, Reference}
+import nl.lumc.sasc.biopet.extensions.gatk
+import nl.lumc.sasc.biopet.utils.config.Configurable
+import org.broadinstitute.gatk.queue.QScript
+
+import scala.collection.mutable.ListBuffer
+
+/**
+  * Created by pjvan_thof on 26-4-17.
+  */
+class GenotypeGvcfs(val parent: Configurable) extends QScript with BiopetQScript with Reference {
+  qscript =>
+
+  def this() = this(null)
+
+  @Input(required = true, shortName = "V")
+  var inputGvcfs: List[File] = Nil
+
+  val namePrefix = config("name_prefix", default = "multisample")
+
+  val maxNumberOfFiles: Int = config("max_number_of_files", default = 10)
+
+  def finalGvcfFile = new File(outputDir, s"$namePrefix.gvcf.vcf.gz")
+  def finalVcfFile = new File(outputDir, s"$namePrefix.vcf.gz")
+
+  /** Init for pipeline */
+  def init(): Unit = {}
+
+  /** Pipeline itself */
+  def biopetScript(): Unit = {
+    inputGvcfs.foreach(inputFiles :+= InputFile(_))
+
+    val combineJob = new CombineJob(finalGvcfFile, outputDir, inputGvcfs)
+    combineJob.allJobs.foreach(add(_))
+
+    val genotype = new gatk.GenotypeGVCFs(this)
+    genotype.variant = List(finalGvcfFile)
+    genotype.out = finalVcfFile
+    add(genotype)
+  }
+
+  class CombineJob(outputFile: File, outputDir: File, allInput: List[File], group: List[Int] = Nil) {
```

Comment:
For me the constructor of a class is simply for creating it, setting the instance variables and that's it. The rest goes into methods. Is there a reason why this code has to be inside the constructor? I'd move it to a recursive method instead.